core/trino-main/src/test/java/io/trino/connector/MockConnector.java

```diff
@@ -745,6 +748,10 @@ public Map<String, Object> getMaterializedViewProperties(ConnectorSession sessio
         @Override
         public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession session, SchemaTableName viewName)
         {
+            MaterializedViewFreshness freshness = getMaterializedViewFreshness.apply(session, viewName);
+            if (freshness != null) {
+                return freshness;
```

Comment:
We def don't like nullable return values and prefer `Optional`s instead.
However, I also don't want the old logic below to remain here for ever, so eventually `getMaterializedViewFreshness` should return `MaterializedViewFreshness` without optional wrapper.

To avoid scope creep, let's use Optional as getMaterializedViewFreshness return type, with a TODO comment to change this, and address in a follow-up (then default getMaterializedViewFreshness will just throw as not implemented).